src/swarm-lang/Swarm/Language/Syntax/Type.hs

```diff
@@ -0,0 +1,156 @@
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+-- |
+-- SPDX-License-Identifier: BSD-3-Clause
+--
+-- Types represeting the surface syntax and terms for Swarm programming language.
+module Swarm.Language.Syntax.Type (
```

Comment:
Personally I find `Type` a confusing name for this module since it sounds like it would be about the syntax of types in the Swarm language itself.  Perhaps  `Swarm.Language.Syntax.AST` ?

Reply:
Makes sense.